######################################### DO NOT EDIT! #########################################
#### This file is automatically generated from data-raw/2-generate_r_code_from_containers.R ####
################################################################################################

#' @title cellTree vem
#' 
#' @description
#' Will generate a trajectory using [cellTree
#' vem](https://doi.org/10.1186/s12859-016-1175-6).
#' 
#' This method was wrapped inside a
#' [container](https://github.com/dynverse/ti_celltree_vem).
#' The original code of this method is available
#' [here](https://bioconductor.org/packages/release/bioc/html/cellTree.html).
#' 
#' @references duVerle, D.A., Yotsukura, S., Nomura, S., Aburatani, H., Tsuda, K.,
#' 2016. CellTree: an R/bioconductor package to infer the hierarchical structure
#' of cell populations from single-cell RNA-seq data. BMC Bioinformatics 17.
#' 
#' @param method LDA inference method to use. Domain: {VEM}. Default: VEM. Format:
#' character.
#' @param sd_filter Standard-deviation threshold below which genes should be
#' removed from the data. Domain: e^U(-4.61, 1.61). Default: 0.5. Format: numeric.
#' @param width_scale_factor A scaling factor for the dynamically-computed
#' distance threshold (ignored if absolute_width is provided). Higher values will
#' result in less branches in the backbone tree, while lower values might lead to
#' a large number of backbone branches. Domain: e^U(-2.30, 4.61). Default: 1.5.
#' Format: numeric.
#' @param outlier_tolerance_factor Proportion of vertices, out of the total number
#' of vertices divided by the total number of branches, that can be left at the
#' end of the backbone tree-building algorithm. Domain: e^U(-9.21, 6.91). Default:
#' 0.1. Format: numeric.
#' @param rooting_method Method used to root the backbone tree. Must be either
#' NULL or one of ‘longest.path’, ‘center.start.group’ or ‘average.start.group’.
#' ‘longest.path’ picks one end of the longest shortest-path between two vertices.
#' ’center.start.group’ picks the vertex in the starting group with lowest
#' mean-square-distance to the others. ‘average.start.group’ creates a new
#' artificial vertex, as the average of all cells in the starting group. If no
#' value is provided, the best method is picked based on the type of grouping and
#' start group information available. Domain: {longest.path, center.start.group,
#' average.start.group, null}. Default: null. Format: character.
#' @param num_topics Number of topics to fit in the model. Domain: U(2, 15).
#' Default: 4. Format: integer.
#' @param tot_iter Numeric parameters (optional) forwarded to the chosen LDA
#' inference method's contol class. Domain: e^U(9.21, 16.12). Default: 1000000.
#' Format: numeric.
#' @param tolerance Numeric parameters (optional) forwarded to the chosen LDA
#' inference method's contol class. Domain: e^U(-16.12, -6.91). Default: 1e-05.
#' Format: numeric.
#' 
#' @keywords method
#' 
#' @return A TI method wrapper to be used together with
#' \code{\link[dynwrap:infer_trajectories]{infer_trajectory}}
#' @export
ti_celltree_vem <- function(
    method = "VEM",
    sd_filter = 0.5,
    width_scale_factor = 1.5,
    outlier_tolerance_factor = 0.1,
    rooting_method = "null",
    num_topics = 4L,
    tot_iter = 1000000L,
    tolerance = 1e-05
) {
  create_ti_method_container(container_id = "dynverse/ti_celltree_vem:v0.9.9")(
    method = method,
    sd_filter = sd_filter,
    width_scale_factor = width_scale_factor,
    outlier_tolerance_factor = outlier_tolerance_factor,
    rooting_method = rooting_method,
    num_topics = num_topics,
    tot_iter = tot_iter,
    tolerance = tolerance
  )
}

