######################################### DO NOT EDIT! #########################################
#### This file is automatically generated from data-raw/2-generate_r_code_from_containers.R ####
################################################################################################

#' @title Inferring a trajectory inference using RaceID / StemID
#' 
#' @description
#' Will generate a trajectory using [RaceID /
#' StemID](https://doi.org/10.1016/j.stem.2016.05.010).
#' 
#' This method was wrapped inside a
#' [container](https://github.com/dynverse/dynmethods/tree/master/containers/raceid_stemid).
#' The original code of this method is available
#' [here](https://github.com/dgrun/RaceID3_StemID2_package).
#' 
#' @references Grün, D., Muraro, M.J., Boisset, J.-C., Wiebrands, K., Lyubimova,
#' A., Dharmadhikari, G., van den Born, M., van Es, J., Jansen, E., Clevers, H.,
#' de Koning, E.J.P., van Oudenaarden, A., 2016. De Novo Prediction of Stem Cell
#' Identity using Single-Cell Transcriptome Data. Cell Stem Cell 19, 266–277.
#' 
#' @param knn integer; Number of nearest neighbors used to infer corresponding
#' cell types in different batches. (default: `10L`; range: from `5L` to `50L`)
#' @param ccor numeric; Correlation coefficient used as a treshhold for
#' determining genes correlated to eachother. (default: `0.4`; range: from `0` to
#' `1`)
#' @param metric discrete; Distances are computed from the filtered expression
#' matrix after optional feature selection, dimensional reduction, and/or
#' transformation (batch correction). (default: `"pearson"`; values: {`"pearson"`,
#' `"spearman"`, `"logpearson"`, `"euclidean"`})
#' @param sat logical; If \code{TRUE}, then the number of clusters is determined
#' based on finding the saturation point of the mean within-cluster dispersion as
#' a function of the cluster number. If \code{FALSE}, then cluster number needs to
#' be given as \code{cln}.
#' @param samp integer; Number of bootstrapping runs for clusterboot (default:
#' `1000L`; range: from `50L` to `10000L`)
#' @param cln integer; Number of clusters to be used. If \code{sat} is
#' \code{TRUE}, this number is inferred by the saturation criterion. (default:
#' `30L`; range: from `10L` to `100L`)
#' @param clustnr integer; Maximum number of clusters for the derivation of the
#' cluster number by the saturation of mean within-cluster-dispersion. (default:
#' `30L`; range: from `10L` to `100L`)
#' @param bootnr integer; Number of booststrapping runs for clusterboot. (default:
#' `50L`; range: from `20L` to `100L`)
#' @param FUNcluster discrete; Clustering method used by RaceID3. (default:
#' `"kmedoids"`; values: {`"kmedoids"`, `"kmeans"`, `"hclust"`})
#' @param probthr numeric; Outlier probability threshold for a minimum of outlg
#' genes to be an outlier cell. This probability is computed from a negative
#' binomial background model of expression in a cluster. (default: `0.001`; range:
#' from `1e-05` to `1L`)
#' @param outminc integer; Minimal transcript count of a gene in a clusters to be
#' tested for being an outlier gene. (default: `5L`; range: from `0L` to `100L`)
#' @param outlg integer; Minimum number of outlier genes required for being an
#' outlier cell. (default: `2L`; range: from `0L` to `100L`)
#' @param outdistquant numeric; Real number between zero and one. Outlier cells
#' are merged to outlier clusters if their distance smaller than the
#' outdistquant-quantile of the distance distribution of pairs of cells in the
#' orginal clusters after outlier removal. (default: `0.95`; range: from `0` to
#' `1`)
#' @param initial_cmd logical; If TRUE, then the t-SNE map computation is
#' initialized with a configuration obtained by classical multidimensional
#' scaling.
#' @param perplexity integer; Perplexity of the t-SNE map. (default: `30L`; range:
#' from `5L` to `100L`)
#' @param cthr integer; Clusters to be included into the StemID2 analysis must
#' contain more than cthr cells. D (default: `5L`; range: from `1L` to `25L`)
#' @param nmode logical; If \code{TRUE}, then a cell of given cluster is assigned
#' to the link to the cluster with the smallest average distance of the knn
#' nearest neighbours within this cluster.
#' @param projcells_knn integer; See \code{nmode}. (default: `3L`; range: from
#' `3L` to `20L`)
#' @param fr logical; Use Fruchterman-Rheingold layout instead of t-SNE for
#' dimensional-reduction representation of the lineage tree.
#' @param pdishuf integer; Number of randomizations of cell positions for which to
#' compute projections of cells on inter-cluster links. (default: `500L`; range:
#' from `10L` to `10000L`)
#' @param fast logical; If \code{TRUE} and \code{nmode} is \code{FALSE} cells will
#' still be assigned to links based on maximum projections but a fast approximate
#' background model will be used to infer significance. The function will do
#' nothing in this case.
#' @param pthr numeric; P-value cutoff for link significance. This threshold is
#' applied for the calculation of link scores reflecting how uniformly a link is
#' occupied by cells. (default: `0.01`; range: from `1e-05` to `1L`)
#' @param scthr numeric; Score threshold for links to be shown in the graph.
#' (default: `0.2`; range: from `0` to `1`)
#' @inheritParams dynwrap::create_ti_method_with_container
#' 
#' @return A TI method wrapper to be used together with
#' \code{\link[dynwrap:infer_trajectories]{infer_trajectory}}
#' @export
ti_raceid_stemid <- function(
    knn = 10L,
    ccor = 0.4,
    metric = "pearson",
    sat = TRUE,
    samp = 1000L,
    cln = 30L,
    clustnr = 30L,
    bootnr = 50L,
    FUNcluster = "kmedoids",
    probthr = 0.001,
    outminc = 5L,
    outlg = 2L,
    outdistquant = 0.95,
    initial_cmd = TRUE,
    perplexity = 30L,
    cthr = 5L,
    nmode = TRUE,
    projcells_knn = 3L,
    fr = FALSE,
    pdishuf = 500L,
    fast = FALSE,
    pthr = 0.01,
    scthr = 0.2,
    config = dynwrap::container_config()
) {
  create_ti_method_with_container(
    image = repo_digests[["dynverse/raceid_stemid"]],
    config = config
  )(
    knn = knn,
    ccor = ccor,
    metric = metric,
    sat = sat,
    samp = samp,
    cln = cln,
    clustnr = clustnr,
    bootnr = bootnr,
    FUNcluster = FUNcluster,
    probthr = probthr,
    outminc = outminc,
    outlg = outlg,
    outdistquant = outdistquant,
    initial_cmd = initial_cmd,
    perplexity = perplexity,
    cthr = cthr,
    nmode = nmode,
    projcells_knn = projcells_knn,
    fr = fr,
    pdishuf = pdishuf,
    fast = fast,
    pthr = pthr,
    scthr = scthr
  )
}

