######################################### DO NOT EDIT! #########################################
#### This file is automatically generated from data-raw/2-generate_r_code_from_containers.R ####
################################################################################################

#' @title Inferring a trajectory inference using StemID
#' 
#' @description
#' Will generate a trajectory using
#' [StemID](https://doi.org/10.1016/j.stem.2016.05.010).
#' 
#' This method was wrapped inside a
#' [container](https://github.com/dynverse/dynmethods/tree/master/containers/stemid).
#' The original code of this method is available
#' [here](https://github.com/dgrun/StemID).
#' 
#' @references Grün, D., Muraro, M.J., Boisset, J.-C., Wiebrands, K., Lyubimova,
#' A., Dharmadhikari, G., van den Born, M., van Es, J., Jansen, E., Clevers, H.,
#' de Koning, E.J.P., van Oudenaarden, A., 2016. De Novo Prediction of Stem Cell
#' Identity using Single-Cell Transcriptome Data. Cell Stem Cell 19, 266–277.
#' 
#' @param clustnr integer; Maximum number of clusters for the computation of the
#' gap statistic or derivation of the cluster number by the saturation criterion.
#' Default is 30. If more major cell types are expected a higher number should be
#' chosen. (default: `30L`; range: from `10L` to `100L`)
#' @param bootnr integer; Number of booststrapping runs for clusterboot. Default
#' is 50 (default: `50L`; range: from `20L` to `100L`)
#' @param metric discrete; The input data are transformed to a distance object.
#' Distances can be computed based on different metrics. Possible values are
#' "pearson", "spearman", "logpearson", "euclidean", "kendall", "maximum",
#' "manhattan", "canberra", "binary" or "minkowski". Default is "pearson". In case
#' of the correlation based methods, the distance is computed as 1 - correlation.
#' K-medoids clustering is performed on this distance object. (default:
#' `"pearson"`; values: {`"pearson"`, `"spearman"`, `"kendall"`, `"euclidean"`,
#' `"maximum"`, `"manhattan"`, `"canberra"`, `"binary"`, `"minkowski"`})
#' @param num_cluster_method discrete; The type of clustering method, can be sat,
#' gap or manual (default: `"sat"`; values: {`"sat"`, `"gap"`, `"manual"`})
#' @param SE.method discrete; The clustering routine calls a modified version of
#' the maxSE function from the cluster package to determine the first local
#' maximum of the gap statistic. By default, we use the method "Tibs2001SEmax" for
#' calling the first local maximum (see specification of maxSE). This method
#' requires that the maximum exceeds the values of its neighbors by a fraction of
#' their standard deviation. This fraction is defined by the parameter SE.factor.
#' All methods defined for the original maxSE function can also be used. (default:
#' `"Tibs2001SEmax"`; values: {`"firstSEmax"`, `"Tibs2001SEmax"`, `"globalSEmax"`,
#' `"firstmax"`, `"globalmax"`})
#' @param SE.factor numeric; Fraction of the standard deviation by which the local
#' maximum is required to differ from the neighboring points it is compared to.
#' Default is 0.25. (default: `0.25`; range: from `0` to `1`)
#' @param B.gap integer; Number of bootstrap runs for the calculation of the gap
#' statistic. Default is 50. (default: `50L`; range: from `20L` to `100L`)
#' @param cln integer; The number of clusters for k-medoids clustering. Default is
#' 0. In this case, the cluster number is determined based on the gap statistic6
#' and do.gap has to be TRUE. (default: `30L`; range: from `20L` to `100L`)
#' @param FUNcluster discrete; The clustering method applied. One of the following
#' methods can be selected: kmedoids, kmeans, hclust. RaceID3 is designed for
#' k-medoids clustering and therefore it is recommended to use only the kmedoids
#' method. Default is kmedoids. (default: `"kmedoids"`; values: {`"kmedoids"`,
#' `"kmeans"`, `"hclust"`})
#' @param dimred_method discrete; The dimensionality reduction method, can be
#' tsne, sammon or tsne_initcmd (default: `"tsne"`; values: {`"tsne"`, `"sammon"`,
#' `"tsne_initcmd"`})
#' @param outminc integer; Expression cutoff for the identification of outlier
#' genes is defined. Default is 5. (default: `0L`; range: from `0L` to `100L`)
#' @param outlg integer; Minimal number of outlier genes required to identify a
#' cell as an outlier. Default is 2. (default: `2L`; range: from `0L` to `100L`)
#' @param probthr numeric; Defines the probability threshold for outlier calling.
#' If the probability of observing a given expression level for a gene in a cell
#' is lower than this cutoff (based on the negative binomial distribution for the
#' calibrated noise model), the cell is considered an outlier for this gene.
#' Default is 10-3. (default: `0.001`; range: from `1e-10` to `0.1`)
#' @param thr_lower integer; Lower probability for which the number of outliers is
#' computed in order to plot the dependence of the number of outliers on the
#' probability threshold (default: `-10`; range: from `-100` to `-1`)
#' @param thr_upper numeric; Upper probability for which the number of outliers is
#' computed in order to plot the dependence of the number of outliers on the
#' probability threshold (default: `-5`; range: from `-100` to `-1`)
#' @param outdistquant numeric; Outlier cells are merged to outlier clusters if
#' their similarity exceeds the outdistquant-quantile of the similarity
#' distribution for all pairs of cells that are together in one of the original
#' clusters. Default is 0.95. (default: `0.95`; range: from `0` to `1`)
#' @param nmode logical; Boolean argument. If nmode is set to TRUE the assignment
#' to inter-cluster links for each cell is not done based on the longest
#' projection, but based on identifying the cluster (other than the cluster the
#' cell belongs to) that contains the nearest neighbor of the cell, i. e. the cell
#' with the most similar transcriptome. The coordinate on the assigned link is
#' still derived based on the projection. Default is FALSE.
#' @param pdishuf integer; Positive integer. This is the number of randomizations
#' to be performed. As a rule of thumb this number should be at least one order of
#' magnitude larger than the desired p-value on the significance of the number of
#' cells on a connection. Default is 2000. (default: `2000`; range: from `100` to
#' `10000`)
#' @param pthr numeric; Positive number. This number corresponds to the p-value
#' threshold, which is used to determine, whether the magnitude of an observed
#' trajectory is significantly larger than observed for the randomized background
#' distribution. This criterion is not used to infer significance of a link, but
#' shown in a graphical representation of the tree (default: `0.01`; range: from
#' `0` to `1e-04`)
#' @param pethr numeric; Positive number. This number corresponds to the p-value
#' threshold, which isused to determine for each link if it is populated by a
#' number of cellssignificantly larger than expected for the randomized background
#' distribution. This p-value threshold determines, which connections are
#' considered validdifferentiation trajectories in the derived lineage tree.
#' (default: `0.01`; range: from `0` to `1e-04`)
#' @param pvalue_cutoff numeric; Upper cutoff of p-value for determining
#' significant edges (default: `0.05`; range: from `0` to `1`)
#' @param linkscore_cutoff numeric; Lower cutoff of the linkscore for determining
#' significant edges (default: `0.2`; range: from `0` to `1`)
#' @inheritParams dynwrap::create_container_ti_method
#' 
#' @return A TI method wrapper to be used together with
#' \code{\link[dynwrap:infer_trajectories]{infer_trajectory}}
#' @export
ti_stemid <- function(
    clustnr = 30L,
    bootnr = 50L,
    metric = "pearson",
    num_cluster_method = "sat",
    SE.method = "Tibs2001SEmax",
    SE.factor = 0.25,
    B.gap = 50L,
    cln = 30L,
    FUNcluster = "kmedoids",
    dimred_method = "tsne",
    outminc = 0L,
    outlg = 2L,
    probthr = 0.001,
    thr_lower = -10,
    thr_upper = -5,
    outdistquant = 0.95,
    nmode = FALSE,
    pdishuf = 2000,
    pthr = 0.01,
    pethr = 0.01,
    pvalue_cutoff = 0.05,
    linkscore_cutoff = 0.2,
    run_environment = NULL
) {
  create_container_ti_method(
    docker_repository = "dynverse/stemid",
    run_environment = run_environment,
    clustnr = clustnr,
    bootnr = bootnr,
    metric = metric,
    num_cluster_method = num_cluster_method,
    SE.method = SE.method,
    SE.factor = SE.factor,
    B.gap = B.gap,
    cln = cln,
    FUNcluster = FUNcluster,
    dimred_method = dimred_method,
    outminc = outminc,
    outlg = outlg,
    probthr = probthr,
    thr_lower = thr_lower,
    thr_upper = thr_upper,
    outdistquant = outdistquant,
    nmode = nmode,
    pdishuf = pdishuf,
    pthr = pthr,
    pethr = pethr,
    pvalue_cutoff = pvalue_cutoff,
    linkscore_cutoff = linkscore_cutoff
  )
}

