######################################### DO NOT EDIT! #########################################
#### This file is automatically generated from data-raw/2-generate_r_code_from_containers.R ####
################################################################################################

#' @title Inferring a trajectory inference using SLICER
#' 
#' @description
#' Will generate a trajectory using
#' [SLICER](https://doi.org/10.1186/s13059-016-0975-3).
#' 
#' This method was wrapped inside a
#' [container](https://github.com/dynverse/ti_slicer).
#' The original code of this method is available
#' [here](https://github.com/jw156605/SLICER).
#' 
#' @references Welch, J.D., Hartemink, A.J., Prins, J.F., 2016. SLICER: inferring
#' branched, nonlinear cellular trajectories from single cell RNA-seq data. Genome
#' Biology 17.
#' 
#' @param kmin integer; Smallest value of k to try (default: `10L`; range: from
#' `2L` to `20L`)
#' @param m integer; Intrinsic dimension of the data. This parameter mainly
#' influences the visualisation of the results. The real intrinsic dimension will
#' be calculated automaticly.  (default: `2L`; range: from `2L` to `20L`)
#' @inheritParams dynwrap::create_ti_method_with_container
#' 
#' @return A TI method wrapper to be used together with
#' \code{\link[dynwrap:infer_trajectories]{infer_trajectory}}
#' @export
ti_slicer <- function(
    kmin = 10L,
    m = 2L
) {
  create_ti_method_with_container(
    image = "dynverse/ti_slicer",
    version = dynmethods::method_versions[["dynverse/ti_slicer"]],
  )(
    kmin = kmin,
    m = m
  )
}

