######################################### DO NOT EDIT! #########################################
#### This file is automatically generated from data-raw/2-generate_r_code_from_containers.R ####
################################################################################################

#' @title SLICER
#' 
#' @description
#' Will generate a trajectory using
#' [SLICER](https://doi.org/10.1186/s13059-016-0975-3).
#' 
#' This method was wrapped inside a
#' [container](https://github.com/dynverse/ti_slicer).
#' The original code of this method is available
#' [here](https://github.com/jw156605/SLICER).
#' 
#' @references Welch, J.D., Hartemink, A.J., Prins, J.F., 2016. SLICER: inferring
#' branched, nonlinear cellular trajectories from single cell RNA-seq data. Genome
#' Biology 17.
#' 
#' @param kmin Smallest value of k to try. Domain: U(2, 20). Default: 10. Format:
#' integer.
#' @param m Intrinsic dimension of the data. This parameter mainly influences the
#' visualisation of the results. The real intrinsic dimension will be calculated
#' automaticly. Domain: U(2, 20). Default: 2. Format: integer.
#' 
#' @keywords method
#' 
#' @return A TI method wrapper to be used together with
#' \code{\link[dynwrap:infer_trajectories]{infer_trajectory}}
#' @export
ti_slicer <- function(
    kmin = 10L,
    m = 2L
) {
  method_choose_backend(
    package_repository = NULL,
    package_name = NULL,
    function_name = NULL,
    package_version = NULL,
    container_id = "dynverse/ti_slicer:v0.9.9.01"
  )(
    kmin = kmin,
    m = m
  )
}

